pkgs/development/tools/prettierd/default.nix

```diff
@@ -0,0 +1,45 @@
+{ lib
+, mkYarnPackage
+, fetchFromGitHub
+, nodejs
+, makeWrapper
+, extraFlags ? [ ]
+, ...
+}:
+mkYarnPackage rec {
+  pname = "prettierd";
+  version = "0.23.4";
+
+  src = fetchFromGitHub {
+    owner = "fsouza";
+    repo = "prettierd";
+    rev = "v${version}";
+    hash = "sha256-GTukjkA/53N9ICdfCJr5HAqhdL5T0pth6zAk8Fu/cis=";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  buildPhase = ''
+    runHook preBuild
+    yarn prepare
+    yarn build
+    runHook postBuild
+  '';
+
+  postInstall = ''
+    wrapProgram "$out/bin/prettierd" --prefix PATH : "${nodejs}/bin" \
+      ${lib.concatStringsSep "\n" (map (a: "--add-flags ${builtins.toJSON a}") extraFlags)}
```

Comment:
Can you explain why `--add-flags` is needed here?
I couldn't spot a mention of it in https://github.com/fsouza/prettierd

Thanks!

Reply:
--add-flags is to wrap the program with flags if the user so chooses to i.e
```
prettierd {
    extraFlags = [ "--flag1" "--boolflag=true" ]; etc. 
}
```

Since prettierd is meant to run as a daemon, I assumed the user might want to pass those flags to a potential systemd service or a container without hassle.

Reply:
Oh whoops, yeah, thanks.

I don't know why I tried to look in `prettierd`'s documentation for that.

Though I do feel like this isn't the right place for that.

Summoning @drupol, who reviewed the previous attempt at #231712